hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -217,6 +218,9 @@
     // not final for testing purposes
     protected MapOperationProvider operationProvider;
 
+    // can be changed dynamically using internal API
+    private boolean failOnIndeterminateOperationState;
```

Comment:
I was thinking about it, but there is also a deeper problem: `MapProxy` for given IMap is a shared object. So the only safe way of using this flag is as follows:

1. each time when we obtain MapProxy via `HazelcastInstance.getMap()`, we have to set `failOnIndeterminateOperationState` to the same value (if we use it)
2. such configured MapProxy object can be kept and used later

If this happens in the same thread, then volatile is redundant.
In case of Jet we have coroutines-like execution that can use many threads, but there should be a barrier somewhere between tasklet switches.
If this is supposed to happen on many ordinary threads, then obtained and configured `MapProxy` object should be assigned to volatile variable.

In each case `failOnIndeterminateOperationState` does not have to be volatile but `setFailOnIndeterminateOperationState` has to be invoked in a way that guarantees visibility to all threads that will use the IMap proxy object (in addition to doing it always after `HazelcastInstance.getMap()`).

You can such usage in https://github.com/hazelcast/hazelcast/pull/23201/files#diff-1a318f9f6c972c5cf5e2e761737d639e0aa80bc54d8b7df4f49c4ba85cbc0368R280 (flag is set in `safeMap` method).

Reply:
@vbekiaris I added comment regarding correct usage of the method